fog/ledger/enclave/impl/src/lib.rs

```diff
@@ -191,6 +196,164 @@ where
+        // global_txo_count = min(global_txo_count) TODO CONFIRM
+        // results = cat(responses.results)
+        // latest_block_version = max(responses.latest_block_version)
+        // max_block_version = max(latest_block_version,
+        // mc_transaction_core::MAX_BLOCK_VERSION
+
+        // TODO no unwraps
+        let num_blocks = shard_query_responses
+            .iter()
+            .map(|query_response| query_response.num_blocks)
+            .min()
+            .unwrap();
+        let global_txo_count = shard_query_responses
+            .iter()
+            .map(|query_response| query_response.global_txo_count)
+            .min()
+            .unwrap();
+        let latest_block_version = shard_query_responses
+            .iter()
+            .map(|query_response| query_response.latest_block_version)
+            .max()
+            .unwrap();
+        // TODO I believe this needs to be implemented in an oblivious way to meet the
+        // security requirements. I'm not 100% sure what an oblivious approach
+        // to this looks like, though. In general this kind of thing needs to be
+        // talked about.
+        let results = shard_query_responses
+            .into_iter()
+            .flat_map(|query_response| query_response.results)
+            .collect();
+        let max_block_version = max(latest_block_version, *MAX_BLOCK_VERSION);
+
+        let client_query_response = CheckKeyImagesResponse {
+            num_blocks,
+            global_txo_count,
+            results,
+            latest_block_version,
+            max_block_version,
+        };
+        let response_plaintext_bytes = mc_util_serial::encode(&client_query_response);
+        let response =
+            self.ake
+                .client_encrypt(&channel_id, &sealed_query.aad, &response_plaintext_bytes)?;
+
+        Ok(response)
+    }
+
+    fn check_key_image_store(
+        &self,
+        msg: EnclaveMessage<NonceSession>,
+        untrusted_key_image_query_response: UntrustedKeyImageQueryResponse,
+    ) -> Result<EnclaveMessage<NonceSession>> {
+        let channel_id = msg.channel_id.clone();
+        let user_plaintext = self.ake.frontend_decrypt(msg)?;
+
+        let req: CheckKeyImagesRequest = mc_util_serial::decode(&user_plaintext).map_err(|e| {
+            log::error!(self.logger, "Could not decode user request: {}", e);
+            Error::ProstDecode
+        })?;
+
+        let mut resp = CheckKeyImagesResponse {
+            num_blocks: untrusted_key_image_query_response.highest_processed_block_count,
+            results: Default::default(),
+            global_txo_count: untrusted_key_image_query_response
+                .last_known_block_cumulative_txo_count,
+            latest_block_version: untrusted_key_image_query_response.latest_block_version,
+            max_block_version: untrusted_key_image_query_response.max_block_version,
+        };
+
+        // Do the scope lock of keyimagetore
```

Comment:
nit: Remove. The code is clear enough and the comment doesn't add much.

Reply:
Addressed in [278d8c1](https://github.com/mobilecoinfoundation/mobilecoin/pull/2896/commits/278d8c1f98696a964ebd09ae87c4b2d5e316cfd0)